File: prime_factors.py
"""
Complexity

PrimeFacors -> O(sqrt(n))
"""


def primeFactors(number: int):
    """return prime numbers wich return number
    when are multipyed"""

    print(f"---{number}---")

    while number % 2 == 0:
        print(2)
        number = number / 2

    # skip one element
    i = 3

    while i * i <= number:

        while number % i == 0:
            print(i)
            number = number / i

        i = i + 2

    # when n is a prime number greather than 2
    if number > 2:
        print(number)


def main():
    for x in range(1, 31, 1):
        primeFactors(x)


if __name__ == '__main__':
    main()

File: prime_factors.c
/**
 * Complexity
 *
 * PrimeFacors -> O(sqrt(n))
 **/

#include <stdio.h>
#include <stdlib.h>

// return prime numbers wich return number when are multipyed
void prime_factors(int number) {

  printf("\n\n---%d---\n", number);

  while (number % 2 == 0) {
    printf("2\n");
    number = number / 2;
  }

  // skip one element
  for (int i = 3; i * i <= number; i = i + 2) {

    while (number % i == 0) {
      printf("%d\n", i);
      number = number / i;
    }
  }

  // when n is a prime number greather than 2
  if (number > 2) {
    printf("%d", number);
  }

  return;
}

int main() {
  for (int x = 1; x <= 30; x++) {
    prime_factors(x);
  }
}
